google/cloud/aiplatform/training_jobs.py

```diff
@@ -1185,4 +1177,207 @@ def _model_upload_fail_string(self) -> str:
+                during Model training. The column must have numeric values between 0 and
+                10000 inclusively, and 0 value means that the row is ignored.
+                If the weight column field is not set, then all rows are assumed to have
+                equal weight of 1.
+            budget_milli_node_hours (int):
+                Optional. The train budget of creating this Model, expressed in milli node
+                hours i.e. 1,000 value in this field means 1 node hour.
+                The training cost of the model will not exceed this budget. The final
+                cost will be attempted to be close to the budget, though may end up
+                being (even) noticeably smaller - at the backend's discretion. This
+                especially may happen when further model training ceases to provide
+                any improvements.
+                If the budget is set to a value known to be insufficient to train a
+                Model for the given training set, the training won't be attempted and
+                will error.
+                The minimum value is 1000 and the maximum is 72000.
+            model_display_name (str):
+                Optional. If the script produces a managed AI Platform Model. The display name of
+                the Model. The name can be up to 128 characters long and can be consist
+                of any UTF-8 characters.
+            disable_early_stopping (bool):
+                Required. If true, the entire budget is used. This disables the early stopping
+                feature. By default, the early stopping feature is enabled, which means
+                that training might stop before the entire training budget has been
+                used, if futrher training does no longer brings significant improvement
+                to the model.
+
+        Returns:
+            model: The trained AI Platform Model resource or None if training did not
+                produce an AI Platform Model.
+
+        Raises:
+            RuntimeError if Training job has already been run
+        """
+
+        training_task_definition = schema.training_job.definition.tabular_task
+
+        training_task_inputs_dict = {
+            # required inputs
+            "targetColumn": target_column,
+            "transformations": self._column_transformations,
+            "trainBudgetMilliNodeHours": budget_milli_node_hours,
+            # optional inputs
+            "weightColumnName": weight_column,
+            "disableEarlyStopping": disable_early_stopping,
+            "optimizationObjective": self._optimization_objective,
+            "predictionType": self._optimization_prediction_type,
+            "optimizationObjectiveRecallValue": self._optimization_objective_recall_value,
+            "optimizationObjectivePrecisionValue": self._optimization_objective_precision_value,
+        }
+
+        model = gca_model.Model(display_name=model_display_name,)
```

Comment:
If the display name is None does the service generate a name for the Model?

Reply:
Ah, I see that the behaviour is different for every datatype. Tables requires a model to be provided according to the yaml and is not actually optional as the protos suggest.